routing.go

```diff
@@ -488,6 +509,10 @@ func (dht *IpfsDHT) FindProviders(ctx context.Context, c cid.Cid) ([]peer.AddrIn
 // Peers will be returned on the channel as soon as they are found, even before
 // the search query completes.
 func (dht *IpfsDHT) FindProvidersAsync(ctx context.Context, key cid.Cid, count int) <-chan peer.AddrInfo {
+	if !dht.enableProviders {
+		return nil
```

Comment:
Consuming from a nil channel will block forever. We either document that the caller must check the return value for a nil channel, or we return a constant 0 length closed channel here. Otherwise, this is a footgun.

Reply:
Thanks for catching this.